onnx_caffe2/backend.py

```diff
@@ -297,7 +299,65 @@ def _create_gemm(cls, n, opset_version):
         return ops
 
     @classmethod
-    def _create_pad(cls, n, opset_version):
+    def _create_lstm(cls, onnx_model, model_helper, n, opset_version):
+        assert model_helper is not None, "cannot convert LSTMs through this codepath"
+        assert onnx_model is not None, "cannot convert LSTMs through this codepath"
+        allowed_attrs = {'hidden_size'}
+        for x in n.attrs.keys():
+            assert x in allowed_attrs, "unsupported attribute " + x + " to LSTM"
+
+        input_blob, W, R, B, sequence_lens, initial_h, initial_c = n.inputs
+
+        hidden_size = n.attrs['hidden_size']
+
+        # ad-hoc, informally-specified, bug-ridden, slow
+        # implementation of shape inference
+        for x in onnx_model.graph.input:
+            if x.name == input_blob:
+                input_size = x.type.tensor_type.shape.dim[2].dim_value
+        for x in map(OnnxNode, onnx_model.graph.node):
+            if x.op_type == n.op_type and x.outputs[0] == input_blob:
+                input_size = x.attrs['hidden_size']
+
+        name = dummy_name()
+
+        # input and recurrence biases are squashed together in onnx but not in caffe2
+        model_helper.net.Slice(B, name + "_bias_i2h",   starts=[0],             ends=[4*hidden_size])
+        model_helper.net.Slice(B, name + "_bias_gates", starts=[4*hidden_size], ends=[8*hidden_size])
+
+        # caffe2 has a different order from onnx. We need to rearrange
```

Comment:
so the plan is to generate ops and rely on constant folding later? Alternative we could manipulate graph initializers explicitly too...

Reply:
as before, unfortunately no, currently the optimization passes don't operate on Caffe2 (though we could add that).